src/dune_console/progress.ml

```diff
@@ -3,25 +3,35 @@ open Stdune
 module No_flush = struct
   let status_line = ref Pp.nop
   let start () = ()
-  let status_line_len = ref 0
+  let status_line_height = ref 0
+  let ansi_erase = "\r\027[2K"
+  let ansi_up = "\027M"
+  let clear_code_length = String.length ansi_erase + String.length ansi_up
 
   let hide_status_line () =
-    if !status_line_len > 0 then Printf.eprintf "\r%*s\r" !status_line_len ""
+    if !status_line_height > 0
+    then (
+      let buf = Buffer.create (clear_code_length * !status_line_height) in
+      Buffer.add_string buf ansi_erase;
+      for _ = 1 to !status_line_height do
+        Buffer.add_string buf (ansi_up ^ ansi_erase)
+      done;
+      Buffer.contents buf |> Printf.eprintf "%s")
   ;;
```

Comment:
should be `prerr_string` along with a flush.

Also, there's no need for the intermediate buffer. `Stdlib.stderr` is already buffered. Just print directly.